File: playwright/PageObjects/HomePage.js
module.exports = class HomePage {
  constructor(page) {
    this.page = page;
  }

  async goToViewSite() {
    return this.page.click('a:text("View site")');
  }

  async goToPosts() {
    return this.page.click('.gh-nav-list-new.relative > a[href="#/posts/"]');
  }

  async classicError() {
    return this.page.click('.gh-nav-list-new.relative > a[href="#/posts/"]');
  }

  async goToPages() {
    return this.page.click('a[href="#/pages/"]');
  }

  async goToPages2() {
    return this.page.click('a[href="#/pages/"]');
  }

  async goToTags() {
    return this.page.click('a[href="#/tags/"]');
  }

  async goToTags2() {
    return this.page.click('a[href="#/tags/"]');
  }

  async closePublishedPostNotification() {
    return this.page.click('.gh-notification-close');
  }

  async signOut() {
    await this.page.click('.gh-user-name.mb1');
    return this.page.click('a[href="#/signout/"]');
  }

  async goToMyProfile() {
    await this.page.click('.gh-user-name.mb1');
    return this.page.click('a:has-text("Your profile")');
  }

  async goToViewSite() {
    return this.page.click('.relative > span > a[href="#/site/"]');
  }

  async getMessageAlertNotification() {
    const notification = await this.page.$$(".gh-alert-content");
    return notification[0].innerText();
  }

  async getMessageAlertNotification2() {
    const notification = await this.page.$$(".gh-alert-content");
    return notification[0].innerText();
  }

  async confirmLeaveCurrentPage() {
    return await this.page.click('.modal-footer > .gh-btn.gh-btn-red');
  }

  async confirmLeaveCurrentPage() {
    return await this.page.click('.modal-footer > .gh-btn.gh-btn-red');
  }

  async confirmLeaveCurrentPage() {
    return await this.page.click('.modal-footer > .gh-btn.gh-btn-red');
  }

  async confirmLeaveCurrentPage() {
    return await this.page.click('.modal-footer > .gh-btn.gh-btn-red');
  }


  async please() {

  }
};

File: kraken/features/web/step_definitions/page_objects/home_page.rb
class HomePage
  attr_reader :driver

  def initialize(driver)
    @driver = driver
  end

  def go_to_posts
    @driver.find_element(
      :css, '.gh-nav-list-new.relative > a[href="#/posts/"]'
    ).click
  end

  def go_to_pages
    @driver.find_element(
      :css, 'a[href="#/pages/"]'
    ).click
  end

  def go_to_tags
    @driver.find_element(
      :css, 'a[href="#/tags/"]'
    ).click
  end

  def close_published_post_notification
    @driver.find_element(
      :css, '.gh-notification-close'
    ).click
  end

  def close_changed_password_notification
    @driver.find_element(
      :css, '.gh-notification-close'
    ).click
  end

  def sign_out
    @driver.find_element(
      :css, '.gh-user-name.mb1'
    ).click
    sleep 1
    @driver.find_element(
      :css, 'a[href="#/signout/"]'
    ).click
  end

  def go_to_my_profile
    @driver.find_element(
      :css, '.gh-user-name.mb1'
    ).click
    sleep 1
    @driver.find_elements(
      :css, 'a'
    ).select do |element|
      element.text == 'Your Profile'
    end.first.click
  end
end
